vcd/resource_vcd_catalog_item.go

```diff
@@ -133,18 +140,97 @@ func resourceVcdCatalogItemCreate(d *schema.ResourceData, meta interface{}) erro
 	d.SetId(catalogName + ":" + itemName)
 
 	log.Printf("[TRACE] Catalog item created: %#v", itemName)
+
+	err = createOrUpdateCatalogItemMetadata(d, meta)
+	if err != nil {
+		return fmt.Errorf("error adding catalog item metadata: %s", err)
+	}
+
 	return resourceVcdCatalogItemRead(d, meta)
 }
 
 func resourceVcdCatalogItemRead(d *schema.ResourceData, meta interface{}) error {
-	return findCatalogItem(d, meta.(*VCDClient))
+	catalogItem, err := findCatalogItem(d, meta.(*VCDClient))
+	if err != nil {
+		return err
+	}
+
+	vAppTemplate, err := catalogItem.GetVAppTemplate()
```

Comment:
Please add a comment above about why this is needed. I mean about your findings, that UI doesn't show metadata of the catalog and you need to actually set it in the child vApp template.

Reply:
added